_adr/87/index.adoc

```diff
@@ -15,31 +15,146 @@ applies_patterns: # What APs does this ADR apply?
 // * No unexpanded acronyms
 // * No undefined jargon
 
-// No need for a title heading, it's added by the template
-
 ## Context and Problem Statement
-// What is the background against which this decision is being taken?
+The KAS Fleet Manager API provides endpoints which end users can use to collect metrics of their own Kafka instances.
+The metrics returned in these endpoints are what we call customer facing metrics.
```

Comment:
It's worth saying that:
* these endpoints are intended to be scraped by Prometheus. I.e. they're not just any old REST API which we can evolve _at will_.
* the Prometheus format doesn't have any built in mechanisms for handling things like versioning or deprecation, and therefore this policy should address these things.
* the Prometheus format does come with guidelines on metric naming, which this policy needs to explicitly either accept, or reject (with reasons for why).

Reply:
> these endpoints are intended to be scraped by Prometheus. I.e. they're not just any old REST API which we can evolve at will.

This is not necessarily true. We do have the `/metrics/query` and `/metrics/query_range` which are used for the Kafka dashboards in the UI and these returns metrics in a JSON format. These endpoints, I think we can evolve at will. The only endpoint we have that's scrapable by Prometheus is `/metrics/federate` and that needs to conform to a Prometheus Text format. I'll make that more clear in this document as well. 

> What competitors are doing is certainly interesting, and might provide some ideas/inspiration, but I don't think we need so much detail in this document.

It's the first time I've done competitive analysis with an ADR so I wasn't sure how much details I needed to put into that. Do you reckon 1/2 examples would've been sufficient?

Reply:
Thanks, that makes sense. Since others were interested in those results, it might be best to record that in the epic/jira issue instead :thinking: 

I'll remove that section from this document :)